enhancements/network/host-port-registry.md

```diff
@@ -0,0 +1,111 @@
+---
+title: Host Port Registry
+authors:
+  - "@squeed"
+reviewers:
+  - "@dcbw"
+  - "@jsafrane"
+  - "@
+approvers:
+  - "@smarterclayton"
+creation-date: 2020-08-26 
+last-updated: 2020-08-26
+status: implementable
+---
+
+# Host Port Registry
+
+OpenShift has a number of host-level services that listen on a socket. Since they
+are not isolated by network namespace, they can conflict if both listen on the same
+port.
+
+This document serves as the authoritative list of port assignments.
+
+## Background
+
+Ports 9000-9999 are available for general use by system services. We require that
+customers open these ports between nodes in a cluster.
+
+These ports are used by certain services that, for whatever reason, must run
+in the host network (or use host-port forwarding).
+
+The Kubernetes scheduler is aware of host ports, and will fail to schedule pods
+on a node where there would be a port conflict.
+
+### Localhost
+
+Separately, it is a common pattern for a pod to have one or more processes only
+listening on localhost. Since all HostNetwork pods share a network namespace,
+coordination is also required in this case.
+
+## Requirements
+
+All processes that wish to listen on a host port MUST
+
+- Have an entry in the table below
+- be in a (user documented range)[https://github.com/openshift/openshift-docs/blob/master/modules/installation-network-user-infra.adoc], 
+  if they are intended to be reachable 
+- declare that port in their `Pod.Spec`
+
+Localhost-only ports SHOULD be outside of this range, to leave room.
+
+# Port Registry
+
+## External
+
+All ports are TCP.
```

Comment:
```suggestion
All ports are TCP unless otherwise specified.
```

Reply:
Also, how about:

> You should consider that these ports will be in use by default unless otherwise specified in the `Notes` section.

I think it would be helpful to call out the conditions where ports will be used or not.  For example, we should call out that the `openshift-sdn` and `ovn-kubernetes` ports depend on the configured network provider.